core/src/Kokkos_Atomics_Desul_Wrapper.hpp

```diff
@@ -53,197 +53,118 @@ inline const char* atomic_query_version() { return "KOKKOS_DESUL_ATOMICS"; }
 #define KOKKOS_DESUL_MEM_SCOPE desul::MemoryScopeDevice()
 #endif
 
-template<class T> KOKKOS_INLINE_FUNCTION
-T atomic_load(T* const dest) { return desul::atomic_load(dest, desul::MemoryOrderRelaxed(), KOKKOS_DESUL_MEM_SCOPE); }
-
-template<class T> KOKKOS_INLINE_FUNCTION
-void atomic_store(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { return desul::atomic_store(dest, val, desul::MemoryOrderRelaxed(), KOKKOS_DESUL_MEM_SCOPE); }
-
-template<class T> KOKKOS_INLINE_FUNCTION
-void atomic_assign(T* const dest, desul::Impl::dont_deduce_this_parameter_t<const T> val) { atomic_store(dest,val); }
+namespace Impl {
+template <class T>
+using not_deduced_atomic_t =
+    std::add_const_t<std::remove_volatile_t<type_identity_t<T>>>;
```

Comment:
The adding the const-qualifier is to match `desul::Impl::dont_deduce_this_parameter_t<const T>`.
I don't think it is particularly useful.